### ISSUE TITLE
Some discord giveaway bots not supported

### ISSUE THREAD

**Issue by Carlos-Espitia** [dfrnoch/discord-sniper#21]
Public giveaway bots:
Plasma
Santa Lunar
Kuroka
Givebot
Gibo | ID: 745054119261372447

Some servers have their own custom giveaway bot, I think it would be nice if you could try to detect if a bot has messaged on a discord with these keywords "giveaway" and detect if the message was from a bot and also detect atleast 1 🎉 emoji with the message to enter the giveaway if that server uses a custom bot.

**Comment by ghost** [comment]
Hey @carlos3212345 I was thinking about that too. The bot is actually looking for the keyword giveaway, but only in plain text messages. Are the giveaways you're talking about in embed form? 
Embed messages look like this:
![maxresdefault](https://user-images.githubusercontent.com/66567067/102220904-bcbd0500-3ee1-11eb-84df-3cd3056faa88.jpg)

**Comment by Carlos-Espitia** [comment]
they are embed messages, also I noticed that some nitro giveaway discords staff would try to ping you if you won one of the giveaways and give you a couple of seconds to dm the hoster of the giveaway. 
![photo_2020-12-15_14-01-01](https://user-images.githubusercontent.com/46390030/102260121-ff7cde00-3edd-11eb-8c87-6b7ebdc3c481.jpg)

I was thinking if yall could add a feature that would detect when the giveaway has finished, detect if our own username is in giveaway "winner: {username}" slot and grab the "Hosted by: {username}" and automatically dm them. The auto dm part could be added in the config file so users could set the dm message when the bot auto dms the hoster of the giveaway.

**Comment by Carlos-Espitia** [comment]
and for the rerolls yall could prob detect the message in that channel and grab the hoster's username and auto dm him

**Comment by ghost** [comment]
that's a good idea! However I think, that the repository owner should take a look at this and decide whether it is useful or not, before anyone of us tries to implement it. But if you want, you can always try to code it yourself and make a pull request.
Besides that I also have to say, that you cant detect edited embed messages with the on_message function the selfbot's using. I don't know if there's any work around, but I'll try to find one soon.

**Comment by Carlos-Espitia** [comment]
@neo-philipp I've implemented some to the auto "DM giveaway host if user wins giveaway" feature but I couldn't figure out how to grab the giveaway hoster's ID to auto DM him. I'm not sure if it's possible to grab other old edited messages from the channel to check who hosted the giveaway.

code:
lines: 409-431
```
    if f'Congratulations <@{Sniper.user.id}>' in message.content or f'<@{Sniper.user.id}> won' in message.content:
        if giveaway_sniper:
            if message.author.id == 294882584201003009 or message.author.id == 673918978178940951 or message.author.id == 582537632991543307 or message.author.id == 396464677032427530 or message.author.id == 649604306596528138:
                print(""
                      f"\n{Fore.GREEN}{time} - Giveaway Won" + Fore.RESET)
                GiveawayInfo()
                if notification:
                    toaster.show_toast("Sniper",
                                       "Giveaway Won! Look into console",
                                       icon_path="./drop.ico",
                                       duration=7)
                if DM_gw_hoster_if_winner:
                    if DM_gw_edelay:
                        sleep(DM_gw_delay)
                        #code
                        print(f"{Fore.GREEN}DM'd Giveaway host with delay of {DM_gw_delay} seconds!")
                    else:
                        #code
                        print(f"{Fore.GREEN}DM'd Giveaway host")
                else:
                    return
        else:
            return
```
![Capture](https://user-images.githubusercontent.com/46390030/102647895-8a571600-4134-11eb-8d40-329d8c4918c0.PNG)

**Comment by Carlos-Espitia** [comment]
btw do yall have a discord for this project?

**Comment by ghost** [comment]
yes i do have discord, my tag is neophilipp#9687, add me if you want

**Comment by FeelsBadMan1** [comment]
There are a lot of Giveawaybots out there. It should be obvious that the Developer cant implement all. However you can take a look  on the soucecode and understand how it sinpe. you then can add own Bots why adding their ID. This should not be a big deal.

**Comment by Lumi6x9** [comment]
guys neophilipp#9687 tag has changed can u please tell me new tag and u want to grab hoster's id right? is it possible for you to grab next message's sender id after you win a giveaway
![image](https://user-images.githubusercontent.com/92018454/148492782-504b2e92-6f36-410f-b82e-fc1701f9d50f.png)
like this i said 'dm me' and u can grab my id

also neophilipp#9687 please join discord.gg/bongocm and mention me there, i am gawr gura/tea, i will not change my profile so please mention or dm me
![image](https://user-images.githubusercontent.com/92018454/148492877-86d4fb96-f0e2-411d-9d89-2cbaecf9623b.png)
